### ISSUE TITLE
ES6 module for PapaParse

### ISSUE THREAD

**Issue by aishwaryasharma14** [DefinitelyTyped/DefinitelyTyped#45797]
While upgrading to Angular 10, using PapaParse gives following warning message - WARNING in C:\parser.component.ts depends on papaparse. CommonJS or AMD dependencies can cause optimization bailouts.

This seems to be because PapaParse is CommonJS module dependency.

I am importing papaparse as follows -
import * as csvParser from 'papaparse';

and using it as -
csvParser.parse(file, {});

Can you help me with an ES6 module that I could use instead? @sandersn  @mrdziuban @spamshaker

**Comment by peterblazejewicz** [comment]
@aishwaryasharma14 

TL;DR;
I believe it does not depends on how you import the module,
It depends if module provides EcmaScript modules for consumption

The NG warnings are correct (expected) and AFAIK one cannot do much with that package (the original one, I'm not describing definition file here for that package). It's an UMD module without a ES6 available source or published JS modules, so yes, modern tree shaking compilers cannot do optimiation during builds. The package information lists only those plain JS, UMD modules:
https://unpkg.com/browse/papaparse@5.2.0/package.json

One could here improve the definition file itself, but it won't help much with compiler warnings in NG.
I believe at the moment there is no there is no other option then add that package to `allowedCommonJsDependencies` to remove that warning.

**Comment by orta** [comment]
Hi thread, we're moving DefinitelyTyped to use [GitHub Discussions](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53377) for conversations the `@types` modules in DefinitelyTyped.

To help with the transition, we're closing all issues which haven't had activity in the last 6 months, which includes this issue. If you think closing this issue is a mistake, please pop into the [TypeScript Community Discord](https://discord.gg/typescript) and mention the issue in the `definitely-typed` channel.
